### ISSUE TITLE
Support batch deletion

### ISSUE THREAD

**Issue by tuancoltech** [ARK-Builders/ARK-Memo#47]
**Problem**: Right now there's no way to delete notes in batch.
We have to to it one by one.

We should support deleting multiple notes at once since it's much more convenient for users.

<img width="402" alt="Screenshot 2024-01-11 at 22 46 27" src="https://github.com/ARK-Builders/ARK-Memo/assets/4281969/3f4e2eb9-ca87-4673-af7e-66f992cec9cd">

**Comment by kirillt** [comment]
Good idea, also we need "delete all versions" feature. Not sure though if it should delete all parents or all parent and all children?

**Comment by tuancoltech** [comment]
> Good idea, also we need "delete all versions" feature. Not sure though if it should delete all parents or all parent and all children?

@kirillt By `versions`, do you mean the notes created by modifying an existing note?
And by `parent`, do you mean the very first note from which `versions` are created?

**Comment by kirillt** [comment]
@tuancoltech yep, children are created by modifying a note and tracked in versions storage. Parent-children relation forms tree-like structure (1 parent can have multiple children, but a child cannot have more than 1 parent). In strict sense, aparent is only immediate parent, in broad sense grandparent is also a parent.

I meant all ancestors in my previous message.

We could also try implementing DAG structure by introducing "merge" feature, which would combine 2 or more notes (parents) into one note (child).

**Comment by tuancoltech** [comment]
> @tuancoltech yep, children are created by modifying a note and tracked in versions storage. Parent-children relation forms tree-like structure (1 parent can have multiple children, but a child cannot have more than 1 parent). In strict sense, a parent is only immediate parent of a child, I meant all ancestors in my previous message.

@kirillt I see. So you was wondering if we should delete all ancestors excluding the latest modification of the note, or all of them including the last modification.

That makes sense. 
From my point of view we should keep the last modified note, as it's normally the most updated one in a way that users really want it to be.

We can do it in another PR as it's not directly related to the purpose of this PR.
